core/src/main/java/com/amplifyframework/core/category/Category.java

```diff
@@ -86,9 +86,11 @@ public synchronized void configure(
             state.set(State.CONFIGURING);
             try {
                 for (P plugin : getPlugins()) {
-                    String pluginKey = plugin.getPluginKey();
-                    JSONObject pluginConfig = configuration.getPluginConfig(pluginKey);
-                    plugin.configure(pluginConfig != null ? pluginConfig : new JSONObject(), context);
+                    if (configureFromDefaultConfigFile()) {
```

Comment:
why do we need this?

Reply:
logging config is not read from amplifyconfiguration.json so this change is added to prevent calling plugin `configure()` for categories where configuration is read from default config file.